File: Source/AlphaTab/Audio/Synth/Midi/Event/SystemCommonEvent.cs
namespace AlphaTab.Audio.Synth.Midi.Event
{
    internal enum SystemCommonTypeEnum
    {
        SystemExclusive = 0xF0,
        MtcQuarterFrame = 0xF1,
        SongPosition = 0xF2,
        SongSelect = 0xF3,
        TuneRequest = 0xF6,
        SystemExclusive2 = 0xF7
    }

    internal abstract class SystemCommonEvent : MidiEvent
    {
        public override int Channel => -1;

        public override MidiEventType Command => (MidiEventType)(Message & 0x00000FF);

        protected SystemCommonEvent(int delta, byte status, byte data1, byte data2)
            : base(delta, status, data1, data2)
        {
        }
    }
}

File: src/audio/midi/event/SystemCommonEvent.ts
import { MidiEvent, MidiEventType } from '@src/audio/midi/event/MidiEvent';

export enum SystemCommonType {
    SystemExclusive = 240,
    MtcQuarterFrame = 241,
    SongPosition = 242,
    SongSelect = 243,
    TuneRequest = 246,
    SystemExclusive2 = 247
}

export class SystemCommonEvent extends MidiEvent {
    public get channel(): number {
        return -1;
    }

    public get command(): MidiEventType {
        return (this.message & 0x00000ff) as MidiEventType;
    }

    protected constructor(delta: number, status: number, data1: number, data2: number) {
        super(delta, status, data1, data2);
    }
}
